services/apps/alcs/src/portal/public/search/application/public-application-search.service.ts

```diff
@@ -0,0 +1,305 @@
+            , "appSearch"."application_uuid"
+            , "appSearch"."application_region_code" 
+            , "appSearch"."file_number"
+            , "appSearch"."applicant"
+            , "appSearch"."local_government_uuid"
+            , "appSearch"."local_government_name"
+            , "appSearch"."application_type_code"
+            , "appSearch"."status"
+            , "appSearch"."date_submitted_to_alc"
+            , "appSearch"."decision_date"
+            , "applicationType"."audit_deleted_date_at"
+            , "applicationType"."audit_created_at"
+            , "applicationType"."audit_updated_by"
+            , "applicationType"."audit_updated_at"
+            , "applicationType"."audit_created_by"
+            , "applicationType"."short_label"
+            , "applicationType"."label"
+            , "applicationType"."code"
+            , "applicationType"."background_color"
+            , "applicationType"."text_color"
+            , "applicationType"."html_description"
+            , "applicationType"."portal_label"
+            `,
+      );
+    return query;
+  }
+
+  private async compileApplicationSearchQuery(searchDto: SearchRequestDto) {
+    let query = this.applicationSearchRepository
+      .createQueryBuilder('appSearch')
+      .where('appSearch.is_draft = false');
+
+    if (searchDto.fileNumber) {
+      query = query
+        .andWhere('appSearch.file_number = :fileNumber')
+        .setParameters({ fileNumber: searchDto.fileNumber ?? null });
+    }
+
+    if (searchDto.portalStatusCode) {
+      query = query.andWhere(
+        "alcs.get_current_status_for_application_submission_by_uuid(appSearch.uuid) ->> 'status_type_code' = :status",
+        {
+          status: searchDto.portalStatusCode,
+        },
+      );
+    }
+
+    if (searchDto.governmentName) {
+      const government = await this.governmentRepository.findOneByOrFail({
+        name: searchDto.governmentName,
+      });
+
+      query = query.andWhere(
+        'appSearch.local_government_uuid = :local_government_uuid',
+        {
+          local_government_uuid: government.uuid,
+        },
+      );
+    }
+
+    if (searchDto.regionCode) {
+      query = query.andWhere(
+        'appSearch.application_region_code = :application_region_code',
+        {
+          application_region_code: searchDto.regionCode,
+        },
+      );
+    }
+
+    query = this.compileApplicationSearchByNameQuery(searchDto, query);
+    query = this.compileApplicationParcelSearchQuery(searchDto, query);
+    query = this.compileApplicationDecisionSearchQuery(searchDto, query);
+    query = this.compileApplicationFileTypeSearchQuery(searchDto, query);
+
+    return query;
+  }
+
+  private compileApplicationDecisionSearchQuery(
+    searchDto: SearchRequestDto,
+    query,
+  ) {
+    if (
+      searchDto.dateDecidedTo !== undefined ||
+      searchDto.dateDecidedFrom !== undefined
+    ) {
+      query = this.joinApplicationDecision(query);
+      // TODO
```

Comment:
I assume this will be addressed in later mrs, correct?

Reply:
Yes